### PR TITLE
Update mlpack to latest stable version.

### PR DESCRIPTION
**Pull request by rcurtin**
Hello there!  Just a package version update.  Hopefully I can get these automated at some point in the near(ish) future. :)

### CODE REVIEW

File: M/mlpack/build_tarballs.jl [hunk 1]
```diff
@@ -137,6 +137,7 @@ products = [
     LibraryProduct("libmlpack_julia_lars", :libmlpack_julia_lars),
     LibraryProduct("libmlpack_julia_linear_regression",
         :libmlpack_julia_linear_regression),
+    LibraryProduct("libmlpack_julia_linear_svm", :libmlpack_julia_linear_svm)
```
Comment (giordano) [inline M/mlpack/build_tarballs.jl:140]: Missing comma:
```suggestion
    LibraryProduct("libmlpack_julia_linear_svm", :libmlpack_julia_linear_svm),
```
:wink:

**Comment by rcurtin** [comment]
Sigh.  Debugging... (looks like boost serialization is not being linked when it needs to be on Windows...)

**Comment by giordano** [comment]
If you're going to make changes, could you please also update the dependencies to
```julia
dependencies = [
    Dependency("boost_jll"),
    Dependency("armadillo_jll"),
    Dependency("OpenBLAS_jll")
]
```
?

**Comment by rcurtin** [comment]
Sure, I can do that. :+1:

File: M/mlpack/build_tarballs.jl [hunk 2]
```diff
@@ -68,6 +68,11 @@ elif [[ $target == *mingw* ]]; then
     fi
     FLAGS+=(-DLAPACK_LIBRARY=${libdir}/libopenblas${arch_ext}.${dlext})
     FLAGS+=(-DBLAS_LIBRARY=${libdir}/libopenblas${arch_ext}.${dlext})
+
+    # The Boost libraries on Windows appear to have been compiled with
+    # _GLIBCXX_USE_CXX_ABI=0, so we do the same here.
+    # https://stackoverflow.com/questions/33394934/converting-std-cxx11string-to-stdstring
+    FLAGS+=(-D_GLIBCXX_USE_CXX11_ABI=0)
```
Comment (giordano) [inline M/mlpack/build_tarballs.jl:75]: That shouldn't be a problem, we're expanding the C++ string ABI for this reason :confused:

Comment (giordano) [reply M/mlpack/build_tarballs.jl:75]: I've cancelled the jobs, we need to rebuild boost :disappointed:

Comment (giordano) [reply M/mlpack/build_tarballs.jl:75]: See #958.  After that is merged and registered, you should just remove this block.

Comment (rcurtin) [reply M/mlpack/build_tarballs.jl:75]: :+1: that sounds good, I'll keep an eye on that.  Thank you for digging into this!

**Comment by rcurtin** [comment]
@giordano you move faster than me, thank you for the push! :+1:

**Comment by rcurtin** [comment]
Thanks again @giordano! :100:

**Comment by giordano** [comment]
You're welcome!

**Pull request merged**